src/Avalonia.Visuals/Media/TextFormatting/TextCollapsingProperties.cs

```diff
@@ -1,23 +1,27 @@
-﻿namespace Avalonia.Media.TextFormatting
+﻿using System.Collections.Generic;
+
+namespace Avalonia.Media.TextFormatting
 {
     /// <summary>
-    /// Properties of text collapsing
+    /// Properties of text collapsing.
     /// </summary>
     public abstract class TextCollapsingProperties
     {
         /// <summary>
-        /// Gets the width in which the collapsible range is constrained to
+        /// Gets the width in which the collapsible range is constrained to.
         /// </summary>
         public abstract double Width { get; }
 
         /// <summary>
-        /// Gets the text run that is used as collapsing symbol
+        /// Gets the text run that is used as collapsing symbol.
         /// </summary>
         public abstract TextRun Symbol { get; }
 
         /// <summary>
-        /// Gets the style of collapsing
+        /// Collapses given text line.
         /// </summary>
-        public abstract TextCollapsingStyle Style { get; }
+        /// <param name="textLine">Text line to collapse.</param>
+        /// <param name="flowDirection">Text flow direction.</param>
+        public abstract IReadOnlyList<TextRun>? Collapse(TextLine textLine, FlowDirection flowDirection);
```

Comment:
Lets just default to FlowDirection.LeftToRight in all cases. Ideally we need to inherit the current run's bidi level but that needs more adjustments I will do later.